airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSourceOperations.java

```diff
@@ -4,17 +4,189 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MySqlSourceOperations extends AbstractJdbcCompatibleSourceOperations<MysqlType> implements SourceOperations<ResultSet, MysqlType> {
 
-public class MySqlSourceOperations extends JdbcSourceOperations {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MySqlSourceOperations.class);
 
+  /**
+   * @param colIndex 1-based column index.
+   */
   @Override
-  protected void putBoolean(final ObjectNode node, final String columnName, final ResultSet resultSet, final int index)
-      throws SQLException {
+  public void setJsonField(final ResultSet resultSet, final int colIndex, final ObjectNode json) throws SQLException {
+    final ResultSetMetaData metaData = (ResultSetMetaData) resultSet.getMetaData();
+    final Field field = metaData.getFields()[colIndex - 1];
+    final String columnName = field.getName();
+    final MysqlType columnType = field.getMysqlType();
+
+    // https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-type-conversions.html
+    switch (columnType) {
+      case BIT -> {
+        if (field.getLength() == 1L) {
+          // BIT(1) is boolean
+          putBoolean(json, columnName, resultSet, colIndex);
+        } else {
+          putBinary(json, columnName, resultSet, colIndex);
+        }
+      }
+      case BOOLEAN -> putBoolean(json, columnName, resultSet, colIndex);
+      case TINYINT, TINYINT_UNSIGNED -> {
+        if (field.getLength() == 1L) {
+          // TINYINT(1) is boolean
+          putBoolean(json, columnName, resultSet, colIndex);
+        } else {
+          putShortInt(json, columnName, resultSet, colIndex);
+        }
+      }
+      case SMALLINT, SMALLINT_UNSIGNED, MEDIUMINT, MEDIUMINT_UNSIGNED -> putInteger(json, columnName, resultSet, colIndex);
+      case INT, INT_UNSIGNED -> {
+        if (field.isUnsigned()) {
+          putBigInt(json, columnName, resultSet, colIndex);
+        } else {
+          putInteger(json, columnName, resultSet, colIndex);
+        }
+      }
+      case BIGINT, BIGINT_UNSIGNED -> putBigInt(json, columnName, resultSet, colIndex);
+      case FLOAT, FLOAT_UNSIGNED -> putFloat(json, columnName, resultSet, colIndex);
+      case DOUBLE, DOUBLE_UNSIGNED -> putDouble(json, columnName, resultSet, colIndex);
+      case DECIMAL, DECIMAL_UNSIGNED -> putBigDecimal(json, columnName, resultSet, colIndex);
+      case DATE -> putDate(json, columnName, resultSet, colIndex);
+      case DATETIME, TIMESTAMP -> putTimestamp(json, columnName, resultSet, colIndex);
+      case TIME -> putTime(json, columnName, resultSet, colIndex);
+      // The returned year value can either be a java.sql.Short (when yearIsDateType=false)
+      // or a java.sql.Date with the date set to January 1st, at midnight (when yearIsDateType=true).
+      // Currently, JsonSchemaPrimitive does not support integer, but only supports number.
```

Comment:
seems like fixing up jsonSchemaPrimitive is high up on the list of things to fix also?

Reply:
Added an issue: https://github.com/airbytehq/airbyte/issues/8722